src/protocol/transactions.rs

```diff
@@ -513,16 +528,48 @@ impl Fee {
         self.fee.ceil() as u64
     }
 
+    #[cfg(test)]
     fn split(&self) -> (u64, u64) {
         let half = self.as_u64() / 2;
         (half as u64, self.as_u64() - half)
     }
 }
 
+/// Subtracts fee fairly from given amounts
+///
+/// We need to consider a few cases:
+/// - If both amounts are >= DUST, they share the fee equally
+/// - If one amount is < DUST, it set to 0 and the other output needs to cover for the fee.
+pub fn subtract_fee(
+    fee: Amount,
+    dust_limit_maker: Amount,
+    dust_limit_taker: Amount,
+    amount_maker: Amount,
+    amount_taker: Amount,
+) -> Result<(Amount, Amount)> {
```

Comment:
🔧 It seems like this belongs with the `Fee` type.

Reply:
done :)